test/gurka/test_route_incidents.cc

```diff
@@ -0,0 +1,685 @@
+#include <gtest/gtest.h>
+
+#include "gurka.h"
+#include "test/util/traffic_utils.h"
+
+using namespace valhalla;
+
+class incidents : public ::testing::Test {
+protected:
+  static gurka::map map;
+
+  static void SetUpTestSuite() {
+    const std::string ascii_map = R"(
+    A----------0----------B----------1----------C
+    |                     |
+    |                     |
+    |                     |
+    2                     3
+    |                     |
+    |                     |
+    |                     |
+    D----------4----------E----------5----------F
+    |                     |
+    |                     |
+    |                     |
+    6                     7
+    |                     |
+    |                     |
+    |                     |
+    G----------8----------H----------9----------I)";
+
+    // connect the ways via the nodes
+    const gurka::ways ways = {
+        {"AB", {{"highway", "tertiary"}}},  {"BC", {{"highway", "tertiary"}}},
+        {"DEF", {{"highway", "primary"}}},  {"GHI", {{"highway", "primary"}}},
+        {"ADG", {{"highway", "motorway"}}}, {"BE", {{"highway", "secondary"}}},
+        {"EH", {{"highway", "service"}}},
+    };
+
+    // generate the lls for the nodes in the map
+    const auto layout = gurka::detail::map_to_coordinates(ascii_map, 10, {.2f, .2f});
+
+    // make the tiles
+    std::string tile_dir = "test/data/route_incidents";
+    map = gurka::buildtiles(layout, ways, {}, {}, tile_dir,
+                            {
+                                {"mjolnir.traffic_extract", tile_dir + "/traffic.tar"},
+                                {"mjolnir.simulate_incidents", "true"},
+                            });
+
+    // stage up some live traffic data
+    valhalla_tests::utils::build_live_traffic_data(map.config);
+  }
+};
+
+// initialize static member
+gurka::map incidents::map = {};
+
+// for asserting where the incident starts and ends on the route and how long it is
+struct incident_location {
+  uint64_t id;
+  uint32_t leg_index;
+  uint32_t begin_edge_index;
+  double begin_pct;
+  uint32_t end_edge_index;
+  double end_pct;
+};
+
+void check_incident_locations(const valhalla::Api& api,
```

Comment:
this is the work horse of the tests. it checks a bunch of stuff to make sure the incidents output is correct. please read over it